include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
```

Comment:
A cast from specific to generic is automatic, provided the generic is a base class of the specific. But here the issue is that the argument to DefaultColorSpaceFunction is taking CompoundData, but the metadata is a CompoundObject. CompoundObject can't be cast to Data because it doesn't derive from Data.

There are two ways of approaching this. The first is just to make the argument to `DefaultColorSpaceFunction` a `const CompoundObject *` instead of `const CompoundData *`. The other is to redefine the metadata plug so that it carries CompoundData instead of CompoundObject. Although it's a bit more work, I think there's a good argument for the latter - it seems unlikely we'd want to use anything more than data in the metadata, and it seems even less likely that a file format would support more than just data either. @andrewkaufman, I think you were keen on using CompoundObject when you first implemented this - still feel that way?